custom/enikshay/integrations/ninetyninedots/tests/test_repeaters.py

```diff
@@ -168,27 +182,105 @@ def test_create_person_no_episode(self):
         self.assertEqual(0, len(self.repeat_records().all()))
 
 
-class TestRegisterPatientPayloadGenerator(ENikshayCaseStructureMixin, TestCase):
+class TestAdherenceRepeater(ENikshayRepeaterTestBase):
+
+    def setUp(self):
+        super(TestAdherenceRepeater, self).setUp()
+        self.repeater = NinetyNineDotsAdherenceRepeater(
+            domain=self.domain,
+            url='case-repeater-url',
+        )
+        self.repeater.white_listed_case_types = ['adherence']
+        self.repeater.save()
+
+    @run_with_all_backends
+    def test_trigger(self):
+        self.create_case_structure()
+        self._create_99dots_registered_case()
+        self._create_99dots_enabled_case()
+        self.assertEqual(0, len(self.repeat_records().all()))
+
+        self.create_adherence_cases([datetime(2017, 02, 17)])
```

Comment:
nit - these are octal `02`'s. really only an issue in august and september though.